lib/widgets/all_channels.dart

```diff
@@ -96,28 +98,34 @@ class AllChannelsListEntry extends StatelessWidget {
     final Subscription? subscription = channel is Subscription ? channel : null;
     final hasContentAccess = store.selfHasContentAccess(channel);
 
-    return Padding(
-      padding: EdgeInsetsDirectional.only(start: 8, end: 4),
-      child: Row(spacing: 6, children: [
-        Icon(
-          size: 20,
-          color: colorSwatchFor(context, subscription).iconOnPlainBackground,
-          iconDataForStream(channel)),
-        Expanded(
-          child: Text(
-            style: TextStyle(
-              color: designVariables.textMessage,
-              fontSize: 17,
-              height: 20 / 17,
-            ).merge(weightVariableTextStyle(context, wght: 600)),
-            channel.name)),
-        if (hasContentAccess) _SubscribeToggle(channel: channel),
-        ZulipIconButton(
-          icon: ZulipIcons.more_horizontal,
-          onPressed: () {
-            showChannelActionSheet(context, channelId: channel.streamId);
-          }),
-      ]));
+    return InkWell(
+      onTap: !hasContentAccess ? null : () => Navigator.push(context,
+        MessageListPage.buildRoute(context: context,
+          narrow: ChannelNarrow(channel.streamId))),
+      onLongPress: () => showChannelActionSheet(context, channelId: channel.streamId),
+      child: ConstrainedBox(constraints: const BoxConstraints(minHeight: 44),
+        child: Padding(
+          padding: EdgeInsetsDirectional.only(start: 8, end: 12),
+          child: Row(spacing: 6, children: [
+            Icon(
+              size: 20,
+              color: colorSwatchFor(context, subscription).iconOnPlainBackground,
+              iconDataForStream(channel)),
+            Expanded(
+              child: Text(
+                maxLines: 1,
+                overflow: TextOverflow.ellipsis,
+                style: TextStyle(
+                  color: designVariables.textMessage,
+                  fontSize: 17,
+                  height: 20 / 17,
+                ).merge(weightVariableTextStyle(context, wght: 600)),
+                channel.name)),
+            IgnorePointer(ignoring: !hasContentAccess,
+              child: AnimatedOpacity(opacity: !hasContentAccess ? 0 : 1,
+                duration: Duration(milliseconds: 200),
+                child: _SubscribeToggle(channel: channel)))
```

Comment:
nit: always use trailing comma, unless `)` is on the same line
```suggestion
                child: _SubscribeToggle(channel: channel))),
```

(This will also slightly reduce the diff in the third / next-to-last commit.)